src/library/export/dlglibraryexport.cpp

```diff
@@ -0,0 +1,202 @@
+    m_pDatabaseDirectoryTextField = make_parented<QLineEdit>();
+    m_pDatabaseDirectoryTextField->setReadOnly(true);
+    m_pMusicDirectoryTextField = make_parented<QLineEdit>();
+    m_pMusicDirectoryTextField->setReadOnly(true);
+
+    // Radio buttons to allow choice between exporting the whole music library
+    // or just tracks in a selection of crates.
+    m_pWholeLibraryRadio = make_parented<QRadioButton>(tr("Entire music library"));
+    m_pWholeLibraryRadio->setChecked(true);
+    this->exportWholeLibrarySelected();
+    connect(m_pWholeLibraryRadio,
+            &QRadioButton::clicked,
+            this,
+            &DlgLibraryExport::exportWholeLibrarySelected);
+    m_pCratesRadio = make_parented<QRadioButton>(tr("Selected crates"));
+    connect(m_pCratesRadio,
+            &QRadioButton::clicked,
+            this,
+            &DlgLibraryExport::exportSelectedCratedSelected);
+
+    // Button to allow ability to browse for the export directory.
+    auto pExportDirBrowseButton = make_parented<QPushButton>(tr("Browse"));
+    connect(pExportDirBrowseButton,
+            &QPushButton::clicked,
+            this,
+            &DlgLibraryExport::browseExportDirectory);
+    auto pExportDirLayout = make_parented<QHBoxLayout>();
+    pExportDirLayout->addWidget(m_pBaseDirectoryTextField);
+    pExportDirLayout->addWidget(pExportDirBrowseButton);
+
+    auto pFormLayout = make_parented<QFormLayout>();
+    pFormLayout->addRow(tr("Base export directory"), pExportDirLayout);
+    pFormLayout->addRow(tr("Engine Prime database export directory"), m_pDatabaseDirectoryTextField);
+    pFormLayout->addRow(tr("Copy music files to"), m_pMusicDirectoryTextField);
+
+    // Buttons to begin the export or cancel.
+    auto pExportButton = make_parented<QPushButton>(tr("Export"));
+    pExportButton->setDefault(true);
+    connect(pExportButton, &QPushButton::clicked, this, &DlgLibraryExport::exportRequested);
+    auto pCancelButton = make_parented<QPushButton>(tr("Cancel"));
+    connect(pCancelButton, &QPushButton::clicked, this, &QDialog::reject);
+
+    // Arrange action buttons at bottom of dialog.
+    auto pButtonBarLayout = make_parented<QHBoxLayout>();
+    pButtonBarLayout->addStretch(1);
+    pButtonBarLayout->addWidget(pExportButton);
+    pButtonBarLayout->addWidget(pCancelButton);
+
+    auto pLayout = make_parented<QGridLayout>();
+    pLayout->setColumnStretch(0, 1);
+    pLayout->setColumnStretch(1, 2);
+    pLayout->addWidget(m_pWholeLibraryRadio, 0, 0);
+    pLayout->addWidget(m_pCratesRadio, 1, 0);
+    pLayout->addWidget(m_pCratesList, 2, 0);
+    pLayout->addLayout(pFormLayout, 0, 1, 3, 1);
+    pLayout->addLayout(pButtonBarLayout, 3, 0, 1, 2);
+
+    setLayout(pLayout);
+    setWindowTitle(tr("Export Library"));
+
+    show();
+    raise();
+    activateWindow();
+}
+
+void DlgLibraryExport::setSelectedCrate(CrateId crateId) {
+    m_pCratesRadio->setChecked(true);
+    exportSelectedCratedSelected();
```

Comment:
I don't understand what this method is supposed to do in this context? With its void/void signature it only exists for the sake of intended side-effects which makes it even harder to understand.